src/compiler/types.ts

```diff
@@ -2704,6 +2710,8 @@ namespace ts {
         ContainsObjectLiteral   = 1 << 22,  // Type is or contains object literal type
         /* @internal */
         ContainsAnyFunctionType = 1 << 23,  // Type is or contains object literal type
+        /* @internal */
+        JsxAttributes           = 1 << 24,  // Jsx attributes type

```

Comment:
My preliminary reaction is that JsxAttributes can't produces types that are very different from normal typescript types, so I expect that this flag is not really needed. But I still need to read the rest of the code.